### DISCUSSION TITLE
How to return Vec<PyAny> from python list?

### DISCUSSION THREAD

**Discussion by AnatolyBuga** [PyO3/pyo3#2684] [Questions] [answered]
My python function needs to return a list of objects, each of which cannot be easily converted to rust objects. [This](https://github.com/pola-rs/polars/blob/master/examples/python_rust_compiled_function/src/ffi.rs) script does the conversion. It's done in the function with signature:
```
pub fn py_series_to_rust_series(series: &PyAny) -> PyResult<Series>
```

ie the function converts PyAny into the object I need which is polars::Series.

My python function however has to return **a list** of such objects (a list of python Series):
```
import polars as pl

def add1 ():
    data = {"a": [1, 2], "b": [3, 4]}
    df = pl.DataFrame(data)
    return df.get_columns() 
```

My question is, since the returned object is python list, I want something like:
```
let r: Vec<PyAny> = app.call0(py)?.extract(py)?;
```

**What is the right way to do it? ie How do I turn the returned value into Vec<PyAny> (or anything which implements iter) so that I can use `py_series_to_rust_series` to convert this object into Vec<Series>**?

**Comment by davidhewitt** [comment, marked as answer]
Try

```rust
let r: Vec<Py<PyAny>> = app.call0(py)?.extract(py)?;
```

**Reply by AnatolyBuga** [reply]
Thanks alot @davidhewitt , don't know how I missed that!
